admin/services/data-access/dfe-signin.data.service.js

```diff
@@ -0,0 +1,66 @@
+'use strict'
+
+const config = require('../../config')
+const logger = require('../log.service').getLogger()
+const bluebird = require('bluebird')
+const jwt = bluebird.promisifyAll(require('jsonwebtoken'))
+const axios = require('axios').default
+
+const service = {
+/**
+ * @description completes user sign in after DfE callback received.
+ * @param {object} user the express request.user object
+ * @returns {Promise<string>} the unique role identifier
+ */
+  getDfeRole: async (user) => {
+    try {
+      const token = await createJwtForDfeApi()
+      const userInfo = await getUserInfoFromDfeApi(token, user)
+      return userInfo.roles[0].code
+    } catch (error) {
+      logger.error(`unable to get dfe role for user:${user.id} error:${error.message}`)
+      throw error
+    }
+  }
+}
+
+/**
+ * @description looks up user role information.
+ * @param {string} token a JWT signed with the dfe API secret
+ * @returns {Promise<object>} the user info object returned from the API
+ */
+const getUserInfoFromDfeApi = async (token, user) => {
+  const serviceId = config.Auth.dfeSignIn.clientId // serves as serviceId also, undocumented
+  const orgId = user.organisation.id
+  const baseUrl = config.Auth.dfeSignIn.userInfoApi.baseUrl
+  const apiUrl = `${baseUrl}/services/${serviceId}/organisations/${orgId}/users/${user.providerUserId}`
+
+  const response = await axios.get(apiUrl, {
+    headers: {
+      Authorization: `Bearer ${token}`
+    }
+  })
+
+  if (response.status === 200) {
+    return response.data
+  } else {
+    logger.error(`dfe API call failed: statusCode:${response.status} - ${response.statusText}`)
+    throw new Error(`unsatisfactory response returned from DfE API. statusCode:${response.status}`)
+  }
+}
+
+/**
+ * @description creates a JWT signed with the api secret using the HS256 algorithm.
+ * @returns {Promise<string>} the signed Json Web Token
+ */
+const createJwtForDfeApi = async () => {
```

Comment:
Why is this method async?  It doesn't await.  The promise will still be returned from the `signAsync` method.

Reply:
fixed